admin/angel/Core.c

```diff
@@ -189,14 +200,33 @@ static void initTunfd(Dict* args, void* vcontext, String* txid, struct Allocator
         struct Pipe* p = Pipe_forFiles(fileno, fileno, ctx->base, &jmp.handler, ctx->alloc);
         p->logger = ctx->logger;
         if (type == FileNo_Type_ANDROID) {
+            Log_info(ctx->logger, "Plumbing AndroidWrapper iface in initTunFd() request [%d] [%d]",
+                     (int)*tunfd, (int)*tuntype);
             struct AndroidWrapper* aw = AndroidWrapper_new(ctx->alloc, ctx->logger);
             Iface_plumb(&aw->externalIf, &p->iface);
             Iface_plumb(&aw->internalIf, &ctx->nc->tunAdapt->tunIf);
         } else {
+            Log_info(ctx->logger, "Plumbing iface in initTunFd() request [%d] [%d]",
+                     (int)*tunfd, (int)*tuntype);
             Iface_plumb(&p->iface, &ctx->nc->tunAdapt->tunIf);
         }
+        Log_info(ctx->logger, "Finished initTunFd() request [%d] [%d]", (int)*tunfd, (int)*tuntype);
+
+        String* path = String_printf(ctx->alloc, "/proc/%d/fd", getpid());
+        DIR* dir = opendir(path->bytes);
+        if (!dir) {
+            Log_warn(ctx->logger, "Unable to opendir() [%s] [%s]", path->bytes, strerror(errno));
+        } else {
+            struct dirent* ent;
+            while ((ent = readdir(dir))) {
+                Log_debug(ctx->logger, "[type=%d] %s/%s", ent->d_type, path->bytes, ent->d_name);
+            }
+            closedir(dir);
+        }
+
```

Comment:
I don't like any of this, we have stuff in libuv which will access files and will not break on Windows even though Windows is a broken platform. The right way to do this is make a FileSystem.h file in util/events and a FileSystem.c file in util/events/libuv and in the C file you can pull in uv.h and then export sane functions out of FileSystem.h which can be used in Core.c
Sorry it's complicated and it sucks but the "standard library" for cjdns is only half written and just dropping this stuff in is a portability nightmare...